assignments/walk_tests_repo.go

```diff
@@ -47,20 +45,20 @@ func match(filename, pattern string) bool {
 
 // readTestsRepositoryContent reads dir and returns a list of assignments and
 // the course's Dockerfile content if there exists a 'tests/scripts/Dockerfile'.
-// Assignments are extracted from 'assignment.yml' files, one for each assignment.
+// Assignments are extracted from 'assignment.json' files, one for each assignment.
 func readTestsRepositoryContent(dir string, courseID uint64) ([]*qf.Assignment, string, error) {
 	files, err := walkTestsRepository(dir)
 	if err != nil {
 		return nil, "", err
 	}
 
-	// Process all assignment.yml files first
+	// Process all assignment files
 	assignmentsMap := make(map[string]*qf.Assignment)
 	for path, contents := range files {
 		assignmentName := filepath.Base(filepath.Dir(path))
-		switch filepath.Base(path) {
```

Comment:
Keep the previous logic with switch statement. We do not need the `filename` as a separate variable.